logical/plugin/pb/backend.proto

```diff
@@ -355,98 +400,133 @@ message StorageEntry {
-    bool JWT = 3;
+	string data = 1;
+	int64 TTL = 2;
+	bool JWT = 3;
 }
 
 message ResponseWrapDataReply {
-    ResponseWrapInfo wrap_info = 1;
-    string err = 2;
+	ResponseWrapInfo wrap_info = 1;
+	string err = 2;
 }
 
 message MlockEnabledReply {
-    bool enabled = 1;
+	bool enabled = 1;
 }
 
 message LocalMountReply {
-    bool local = 1;
+	bool local = 1;
 }
 
+// SystemView exposes system configuration information in a safe way for plugins
+// to consume. Plugins should implement the client for this service.
 service SystemView {
-    rpc DefaultLeaseTTL(Empty) returns (TTLReply);
-    rpc MaxLeaseTTL(Empty) returns (TTLReply);
-    rpc SudoPrivilege(SudoPrivilegeArgs) returns (SudoPrivilegeReply);
-    rpc Tainted(Empty) returns (TaintedReply);
-    rpc CachingDisabled(Empty) returns (CachingDisabledReply);
-    rpc ReplicationState(Empty) returns (ReplicationStateReply);
-    rpc ResponseWrapData(ResponseWrapDataArgs) returns (ResponseWrapDataReply);
-    rpc MlockEnabled(Empty) returns (MlockEnabledReply);
-    rpc LocalMount(Empty) returns (LocalMountReply);
+  	// DefaultLeaseTTL returns the default lease TTL set in Vault configuration
+	rpc DefaultLeaseTTL(Empty) returns (TTLReply);
+	
+   	// MaxLeaseTTL returns the max lease TTL set in Vault configuration; backend
+	// authors should take care not to issue credentials that last longer than
+	// this value, as Vault will revoke them
+	rpc MaxLeaseTTL(Empty) returns (TTLReply);
+
+	// SudoPrivilege returns true if given path has sudo privileges
+	// for the given client token
+	rpc SudoPrivilege(SudoPrivilegeArgs) returns (SudoPrivilegeReply);
+
+	// Returns true if the mount is tainted. A mount is tainted if it is in the
+	// process of being unmounted. This should only be used in special
+	// circumstances; a primary use-case is as a guard in revocation functions.
+	// If revocation of a backend's leases fails it can keep the unmounting
+	// process from being successful. If the reason for this failure is not
+	// relevant when the mount is tainted (for instance, saving a CRL to disk
+	// when the stored CRL will be removed during the unmounting process
+	// anyways), we can ignore the errors to allow unmounting to complete.
+	rpc Tainted(Empty) returns (TaintedReply);
+
+	// Returns true if caching is disabled. If true, no caches should be used,
+	// despite known slowdowns.
+	rpc CachingDisabled(Empty) returns (CachingDisabledReply);
+
+   	// ReplicationState indicates the state of cluster replication
+	rpc ReplicationState(Empty) returns (ReplicationStateReply);
+
+	// ResponseWrapData wraps the given data in a cubbyhole and returns the
+	// token used to unwrap.
+	rpc ResponseWrapData(ResponseWrapDataArgs) returns (ResponseWrapDataReply);
+
+	// MlockEnabled returns the configuration setting for enabling mlock on
+	// plugins.
+	rpc MlockEnabled(Empty) returns (MlockEnabledReply);
+
+	// When run from a system view attached to a request, indicates whether the
```

Comment:
"LocalMount, when run from a system view attached to a request, indicates whether ..."